web/source/kmwcallback.ts

```diff
@@ -1185,6 +1195,32 @@ class KeyboardInterface {
   }
   // I3318 - deadkey changes END
 
+  doInputEvent(_target: HTMLElement|Document) {
+      var event: Event;
+      // TypeScript doesn't yet recognize InputEvent as a type!
+      if(typeof window['InputEvent'] == 'function') {
+        event = new window['InputEvent']('input', {"bubbles": true, "cancelable": false});
+      } // No else - there is no supported version in some browsers.
+
+      // Ensure that touch-aliased elements fire as if from the aliased element.
+      if(_target['base'] && _target['base']['kmw_ip']) {
+        _target = _target['base'];
+      }
+
+      if(_target && event) {
+        _target.dispatchEvent(event);
+      }
+  }
+
+  defaultBackspace(Pelem?: HTMLElement|Document) {
+    if(!Pelem) {
+      Pelem = this.keymanweb.domManager.getLastActiveElement();
+    }
+
+    this.output(1, this.keymanweb.domManager.getLastActiveElement(), "");
```

Comment:
Does this work correctly with SMP?

Reply:
Yes.  This is exactly how the recent SMP fix represents a backspace; the `output` function is highly SMP-aware.

Reply:
Great, just checking :)